builder/simplify.cc

```diff
@@ -485,16 +486,20 @@ class simplifier : public node_mutator {
         it = std::make_reverse_iterator(lets.erase(std::next(it).base()));
         values_changed = true;
       } else if (should_substitute(it->second)) {
-        body = mutate(substitute(body, it->first, it->second), &body_bounds);
+        body = substitute(body, it->first, it->second);
```

Comment:
Do we need a comment here to point out that omitting the mutate is deliberate?

Reply:
Eh it's borderline IMO, the recursive mutate call is just a few lines below. Also, *not* recursively mutating is now the default after this PR.